proposed/0000-name-service.md

```diff
@@ -0,0 +1,67 @@
+- Feature Name: Decentralised Naming System (dns)
+- Type new feature
+- Related components maidsafe_client, maidsafe_vault
+- Start Date: 27-06-2015
+- RFC PR:
+- Issue number: 
+
+# Summary
+
+In the current Internet Domains are referenced via a naming system which comprises an indexing mechanism
+This indexing mechanism is centralised and controlled, with the ability for countries and ISPs to switch it off
+pollute data with advertising data and much worse. The SAFE network on the other had offers a 'free (as in beer)'
+mechanism to locate data linked to a name. This proposal outlines a mechanism to look up data related to any name.
+
+This data includes, long name (could be real name is users wish), web site id, blog id, micro-blog id and may be 
+extended to include additional information as the network develops. 
+
+# Motivation
+
+In networks people expect to be able to lookup services related to names. In the SAFE network this includes
+the ability to retrieve public keys to encrypt data to that id, in cases where privacy is a requirement (.e.
+in SAFE all messages are encrypted between identities). The opportunity for people to create and link web sites
+and other services is also a motivation for implementing such a system. 
+
+# Detailed design
+
+This is a simple use case for `Unified Structured Data` (see RFC XXXX). In this case we require to set three items
+
+1. The `type_tag` shall be type `4`
+
+2. The `Identity` shall be the `Sha512` of the publicly chosen name (user chooses this name)
+
+3. The `data` field:
+
+```rust
+struct dns {
+long_name: String
+encrytion_key: crypto::encrypt::public_key
+//      service, location
+HashMap<String, Identity>
+}
+
+Initially this `HashMap` will likely contain www, blog and micro-blog. Application developers may add services to enhance 
+users experiences with new applications and functions via this mechanism. 
+
```

Comment:
So I think entries will be as you say, but found by looking for `SHA512(name + 3)` and ask fro type 3 data (i.e. the public dir). That will give you the `Directory` you want. 

1. Long name is really your name. So this `dns` packet would have `tag_type = 5` and `Identity = SHA512(spandan)` and if I look for your web site for instance I `Get(SHA512(spandan + 5)` and read the www entry (we demand at least www, blog, micro-blog etc. exist so these are required, developers can add more services, but we can set a minimum, tbd). So you will have one web site per identity by default, but you could have more if you add them as services and apps recognised those services. This is probably for later though. To access this all a client (person) will so is type `safe::spandan/www` or `safe::spandan/blog`into their web browser. Same for other services. This can be certainly improved (default to www for instance) 

2. This is currently un-used and can be removed for now. I will make that change. (it'e too much for now). 

3. We can restrict DNS entries if we wish, it is an implementation issue for sure. Good to get community feedback on this one. Personally I think we should restrict to 3 at a time (so require a delete of an old one to create a new one after that)